pkg/asset/cluster/tfvars.go

```diff
@@ -741,6 +741,9 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		if err != nil {
 			return errors.Wrapf(err, "failed to list avaliable zones for NAT gateway")
 		}
+		if len(natGatewayZones.Zones) == 0 {
+			return errors.Wrapf(err, "enhanced NAT gateway is not supported in the region %s", installConfig.AlibabaCloud.Region)
+		}
```

Comment:
Why is this needed here as well as in the machine configs? Isn't it sufficient to check once?

Reply:
After I added this check, I realized it would be more appropriate to add a validation, but I'm not sure if it's superfluous, so I kept it

Reply:
Yeah you can remove this entire check from here. The install config validation will cover it.

Reply:
Have removed